dspace-api/src/main/java/org/dspace/content/dao/impl/MetadataValueDAOImpl.java

```diff
@@ -70,7 +72,13 @@ public void deleteByMetadataField(Context context, MetadataField metadataField)
     public MetadataValue getMinimum(Context context, int metadataFieldId)
             throws SQLException
     {
-        String queryString = "SELECT m FROM MetadataValue m JOIN FETCH m.metadataField WHERE m.metadataField.id = :metadata_field_id ORDER BY text_value";
+	String queryString;
+	if(DatabaseUtils.getDbType().equals(DatabaseUtils.DBMS_ORACLE)){
+	    queryString = "SELECT m FROM MetadataValue m JOIN FETCH m.metadataField WHERE m.metadataField.id = :metadata_field_id ORDER BY DBMS_LOB.substr(text_value , 4000 , 1)";
+	}else{
```

Comment:
With the switch to using Hibernate in DSpace 6, we no longer recommend the (old) model of having separate queries per database type (with an `if` statement surrounding them). Syntaxes for different database types is something that Hibernate should handle for us.  So, if the default query does not work across various database types, then we need to find a Hibernate query that will.

Reply:
I too dislike making special cases per DBMS brand, but it seems necessary in this case.  Yes, it would be nice if Hibernate would just take care of this, but there is a problem:  Oracle will not do comparisons with LOB fields, and limits the various *CHAR types to 2000 or 4000 bytes.  So e.g. converting CLOB to VARCHAR potentially lops off the tail of the value, leading to an inexact comparison.  I can well imagine that the Hibernate maintainers decided to punt if they couldn't give exact results, and it seems that they did.